simple_agent/infrastructure/textual/autocomplete_strategies.py

```diff
@@ -0,0 +1,110 @@
+
+class AutocompleteStrategy(Protocol):
+    def check(self, text_area: TextArea, row: int, col: int, line: str) -> Optional[CompletionContext]:
+        """
+        Check if autocomplete should be triggered.
+        Returns context if triggered, None otherwise.
+        """
+        ...
+
+    async def get_suggestions(self, context: CompletionContext) -> List[Any]:
+        """
+        Get list of suggestions based on context.
+        """
+        ...
+
+    def format_suggestion(self, suggestion: Any) -> str:
+        """
+        Format a suggestion for display in the popup.
+        """
+        ...
+
+    def apply_completion(self, text_area: TextArea, suggestion: Any, context: CompletionContext, row: int, col: int) -> None:
+        """
+        Apply the selected suggestion to the text area.
+        """
+        ...
+
+
+class SlashCommandStrategy:
+    def __init__(self, registry: SlashCommandRegistry):
+        self.registry = registry
+
+    def check(self, text_area: TextArea, row: int, col: int, line: str) -> Optional[CompletionContext]:
+        if row == 0 and col > 0 and line.startswith("/") and " " not in line[:col]:
+            return CompletionContext(query=line[:col], start_index=0, trigger_char="/")
+        return None
+
+    async def get_suggestions(self, context: CompletionContext) -> List[Any]:
+        return self.registry.get_matching_commands(context.query)
+
+    def format_suggestion(self, suggestion: Any) -> str:
+        return f"{suggestion.name} - {suggestion.description}"
+
+    def apply_completion(self, text_area: TextArea, suggestion: Any, context: CompletionContext, row: int, col: int) -> None:
+        cmd_name = suggestion.name
+        text_area.text = cmd_name + " "
+        text_area.move_cursor_relative(columns=len(cmd_name) + 1)
+
+
+class FileSearchStrategy:
+    def __init__(self, searcher: FileSearcher):
+        self.searcher = searcher
+
+    def check(self, text_area: TextArea, row: int, col: int, line: str) -> Optional[CompletionContext]:
+        # Check for file search (@)
+        text_before_cursor = line[:col]
+
+        # Find the start of the word before cursor
+        last_space_index = text_before_cursor.rfind(" ")
+        word_start_index = last_space_index + 1
+        current_word = text_before_cursor[word_start_index:]
+
+        if current_word.startswith("@"):
+             return CompletionContext(query=current_word[1:], start_index=word_start_index, trigger_char="@")
+        return None
+
+    async def get_suggestions(self, context: CompletionContext) -> List[Any]:
+        try:
+            return await self.searcher.search(context.query)
+        except Exception as e:
+            logger.error(f"File search failed: {e}")
+            return []
+
+    def format_suggestion(self, suggestion: Any) -> str:
+        return str(suggestion)
+
+    def apply_completion(self, text_area: TextArea, suggestion: Any, context: CompletionContext, row: int, col: int) -> None:
```

Comment:
I don't like that we pass textarea here. It seems like a completion can yield attachments. Instead of manipulating the textarea we should return the artifacts that are interresting. Like the text that will be inserted, and the attachments that will be attached. The caller will then to the replacement and attaching.